google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractResultSet.java

```diff
@@ -983,11 +1001,15 @@ public boolean isWithBeginTransaction() {
     protected final PartialResultSet computeNext() {
       PartialResultSet next;
       try {
-        // TODO: Ideally honor io.grpc.Context while blocking here.  In practice,
-        //       cancellation/deadline results in an error being delivered to "stream", which
-        //       should mean that we do not block significantly longer afterwards, but it would
-        //       be more robust to use poll() with a timeout.
-        next = stream.take();
+        if (streamWaitTimeoutUnit != null) {
+          next = stream.poll(streamWaitTimeoutValue, streamWaitTimeoutUnit);
+          if (next == null) {
+            throw SpannerExceptionFactory.newSpannerException(
+                ErrorCode.DEADLINE_EXCEEDED, "stream wait timeout");
+          }
+        } else {
+          next = stream.take();
+        }
```

Comment:
The default stream wait timeout that is set for all call contexts is 30 mins. This value can be overridden by configuring a custom call context for a specific query. So, just wanted to understand when will we encounter this else condition (`streamWaitTimeoutUnit == null`) better.

I can think of a case where user explicitly sets a `0` stream wait timeout value in which case the `streamWaitTimeoutUnit` will be `null` here, and IMO, we should directly throw a `DEADLINE_EXCEEDED` error without making any `stream.take()` call.

Reply:
A user could do one of the following:
1. Configure a custom call context with a null stream timeout by calling `context.withStreamWaitTimeout(null)`. That would however in our specific setup not cause a null timeout to be used, because we merge the default context with the call-specific context. Any null values on the call-specific context will not overwrite any of the defaults.
2. As you mentioned: Setting a zero timeout will cause the `streamWaitTimeoutUnit` to be `null`. That is intentional, as the `streamWaitTimeout` should be considered disabled if set to zero. See https://github.com/googleapis/sdk-platform-java/blob/f9d920d076db36dc095f7ace9c571301122ce959/gax-java/gax/src/main/java/com/google/api/gax/rpc/ApiCallContext.java#L95

Reply:
I've added a test to verify this behavior, and some comments to clarify.

Reply:
Thanks for clarifying!